packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -178,27 +215,61 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     return input.files?.first;
   }
 
+  List<html.File>? _getFilesFromInput(html.FileUploadInputElement input) {
+    if (_hasOverrides) {
+      return _overrides!.getMultipleFilesFromInput(input);
+    }
+    return input.files;
+  }
+
   /// Handles the OnChange event from a FileUploadInputElement object
-  /// Returns the objectURL of the selected file.
-  String? _handleOnChangeEvent(html.Event event) {
+  /// Returns the selected file.
+  html.File? _handleOnChangeEvent(html.Event event) {
     final html.FileUploadInputElement input =
         event.target as html.FileUploadInputElement;
-    final html.File? file = _getFileFromInput(input);
+    return _getFileFromInput(input);
+  }
 
-    if (file != null) {
-      return html.Url.createObjectUrl(file);
-    }
-    return null;
+  /// Handles the OnChange event from a FileUploadInputElement object
+  /// Returns the list of selected files.
+  List<html.File>? _handleOnChangeEventForMultipleFiles(html.Event event) {
+    final html.FileUploadInputElement input =
+        event.target as html.FileUploadInputElement;
+    return _getFilesFromInput(input);
   }
 
   /// Monitors an <input type="file"> and returns the selected file.
   Future<PickedFile> _getSelectedFile(html.FileUploadInputElement input) {
     final Completer<PickedFile> _completer = Completer<PickedFile>();
     // Observe the input until we can return something
     input.onChange.first.then((event) {
-      final objectUrl = _handleOnChangeEvent(event);
-      if (!_completer.isCompleted && objectUrl != null) {
-        _completer.complete(PickedFile(objectUrl));
+      final pickedFile = _handleOnChangeEvent(event);
+      if (!_completer.isCompleted && pickedFile != null) {
+        _completer.complete(PickedFile(html.Url.createObjectUrl(pickedFile)));
+      }
+    });
+    input.onError.first.then((event) {
+      if (!_completer.isCompleted) {
+        _completer.completeError(event);
+      }
+    });
+    // Note that we don't bother detaching from these streams, since the
+    // "input" gets re-created in the DOM every time the user needs to
+    // pick a file.
+    return _completer.future;
+  }
+
+  Future<List<PickedFile>> _getSelectedFiles(
```

Comment:
(Please, do not add any new methods that return PickedFile)

Reply:
removed this too .